components/prism-css-extras.js

```diff
@@ -7,9 +7,32 @@
 		pattern: Prism.languages.css.selector,
 		inside: selectorInside = {
 			'pseudo-element': /:(?:after|before|first-letter|first-line|selection)|::[-\w]+/,
-			'pseudo-class': /:[-\w]+/,
+			'pseudo-class': {
+				pattern: /:[-\w]+(?:(?:\([^\(]*\([^\)]*\)[^\)]*\))|(?:\(.*?\)))?/,
+				greedy: true,
+				inside: {
+					'n-th': [
+						{
+							pattern: /(\(\s*)[+-]?\d*[\dn](?:\s*[+-]\s*\d+)?(?=\s*\))/,
+							lookbehind: true,
+							inside: {
+								'number': /[\dn]+/,
+								'operator': /[+-]/
+							}
+						},
+						{
+							pattern: /(\(\s*)(?:even|odd)(?=\s*\))/i,
+							lookbehind: true
+						}
+					],
+					'punctuation': /[()]/,
+				}
+			},
 			'class': /\.[-:.\w]+/,
 			'id': /#[-:.\w]+/,
+			'tag': /[-\w]+/,
```

Comment:
I don't know if I like that. It's consistent with markup but it also completely changes the look of CSS selectors. 
Maybe just give it a different name?